lib/events/dynamoevents/dynamoevents.go

```diff
@@ -548,6 +549,15 @@ func (l *Log) SearchEvents(fromUTC, toUTC time.Time, filter string, limit int) (
 					break
 				}
 			}
+			if fields.GetType() == events.SessionStartEvent && events.IsRecordOff(fields) {
```

Comment:
This chunk of code seems largely identical across the backend implementations, could we somehow factor it out into a backend agnostic variant to keep it DRY?